flink-runtime/src/main/java/org/apache/flink/runtime/state/StateUtil.java

```diff
@@ -70,9 +70,11 @@ public static void bestEffortDiscardAllStateObjects(
      *
      * @param stateFuture to be discarded
      * @throws Exception if the discard operation failed
+     * @return the size of state before cancellation (if available)
      */
-    public static void discardStateFuture(Future<? extends StateObject> stateFuture)
+    public static long discardStateFuture(Future<? extends StateObject> stateFuture)
             throws Exception {
+        long stateSize = 0;
```

Comment:
It would be problematic to combine `-1` with valid values.
I think we should just mention in the docs that this number is best effort if checkpoint isn't completed.